server/py/services/api/api/endpoints/pipelines.py

```diff
@@ -189,7 +189,10 @@ async def retry_pipeline(
 
     # If direct mode is requested, or the original workflow runner was not found,
     # bypass MLRun's workflow runner logic and submit the retry directly to KFP.
-    if submit_mode == mlrun_constants.WorkflowSubmitMode.direct or not original_runner:
+    if (
+        submit_mode == mlrun_constants.WorkflowSubmitMode.direct
+        or not original_runner.spec.notifications
+    ):
```

Comment:
you may have an exception here:
```suggestion
    if (
        submit_mode == mlrun_constants.WorkflowSubmitMode.direct
        or not original_runner
        or not original_runner.spec.notifications
    ):
```